python/feathub/processors/flink/table_builder/redis_utils.py

```diff
@@ -96,9 +107,125 @@ def append_physical_key_columns_per_feature(
     return table, physical_key_column_names
 
 
+def optimize_redis_source_lookup_map(
+    t_env: StreamTableEnvironment,
+    table: NativeFlinkTable,
+    table_descriptor: TableDescriptor,
+    join_field_descriptors: Dict[str, JoinFieldDescriptor],
+) -> NativeFlinkTable:
+    """
+    If the right table of a lookup join is a RedisSource and only designated keys
```

Comment:
Does this mean that we do not support filtering-by-map-key when there are columns with non-map type?

Reply:
Filtering by map key is still supported when there are non-map-typed columns. There might be some ambiguity in this description. I'll improve this document.